### PR TITLE
backend: replace go-bindata with embed

### PR DESCRIPTION
**Pull request by benma**
embed is the new (since Go 1.16) official way to embed static files.

With this commit, the size of libserver.so increases from 20607776
bytes to 20982216, which is roughly 300kB.

One advantage is that missing files or filename typos will be caught
at compile time instead of at runtime.

### CODE REVIEW

**Comment by benma** [comment]
@x1ddos unclear if this change is worth it due to the binary size increase. I just put it up since I had it working so we can see it and come back to it.

**Comment by benma** [comment]
Looking at the deleted assets.go file, it looks like go-bindata was using compression all along:

image

This doesn't seem to exist with embed.FS, but could be accomplished manually :thinking:

**Comment by benma** [comment]
@x1ddos I added a commit that manually decompresses and get back the ~300kB. Will write a test.

**Review by x1ddos — COMMENTED** [review]
I like it. One more dependency crossed out. :+1: 

Not sure I'd even bother with compression. The desktop app size is over 100Mb, Android 60Mb+. 300k is unnoticeable.

Should be easy to test: get the bytes, hash it and compare with a known value? Would be very nice if the hashes were the same as those in assertion.txt files from the firmware repo.

**Comment by benma** [comment]
@x1ddos PTAL. I removed the panics and added some tests.

I dislike tests that need to change frequently, so I only check the magic bytes that are static.

File: backend/devices/bitbox02bootloader/firmware.go [hunk 1]
```diff
@@ -15,27 +15,37 @@
 package bitbox02bootloader
 
 import (
-	"fmt"
+	// Needed for the go:embed directives below.
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/devices/bitbox02bootloader/firmware.go]: oops of course

**Comment by benma** [comment]
> Not sure I'd even bother with compression. The desktop app size is over 100Mb, Android 60Mb+. 300k is unnoticeable.

I did it anyway :innocent: 300kb is 300kb, it adds up :thinking:

**Comment by x1ddos** [comment]
> I dislike tests that need to change frequently, so I only check the magic bytes that are static.

It would prevent mistakes like compressing wrong file. When you update assets/ next time, who's going to double-check the firmare bins are actually correct? If a hash matched that of assertion, it's very easy to verify without even cloning pull request.

Hashes don't have to be in a _test.go file. They could live alongside the firmare.bin in the assets/ dirs. The test would simply read a sha256.txt file and match.

**Comment by benma** [comment]
> > I dislike tests that need to change frequently, so I only check the magic bytes that are static.
> 
> It would prevent mistakes like compressing wrong file. When you update assets/ next time, who's going to double-check the firmare bins are actually correct? If a hash matched that of assertion, it's very easy to verify without even cloning pull request.
> 
> Hashes don't have to be in a _test.go file. They could live alongside the firmare.bin in the assets/ dirs. The test would simply read a sha256.txt file and match.

I see. Good idea, done.

The rest is figuring out how to make Go 1.16 work on macOS CI.

**Review by x1ddos — APPROVED** [review]
LGTM. Very nice.

**Pull request merged**